File: apps/web/lib/api/health.ts
import { apiClient } from "./client";

export interface SystemTelemetry {
  uptime_seconds: number;
  memory_resident_mb: number;
  python_version: string;
  platform: string;
  active_runs_in_memory: number;
  loaded_datasets_count: number;
}

export interface HealthResponse {
  status: string;
  timestamp: string;
  service: string;
  version: string;
  telemetry?: SystemTelemetry;
}

export interface ReadyResponse {
  ready: boolean;
  status: string;
  service: string;
  pipeline_version: string;
  database: string;
  cache: string;
  engines: Record<string, boolean>;
  timestamp: string;
}

export interface VersionResponse {
  api_version: string;
  pipeline_version: string;
  scoring_config_version: string;
  detector_config_version?: string;
  git_commit: string;
  engines: Record<string, string>;
}

export async function getHealth(): Promise<HealthResponse> {
  return apiClient<HealthResponse>("/healthz");
}

export async function getReady(): Promise<ReadyResponse> {
  return apiClient<ReadyResponse>("/readyz");
}

export async function getVersion(): Promise<VersionResponse> {
  return apiClient<VersionResponse>("/api/v1/version");
}

File: app/api/v1/health.py
from fastapi import APIRouter
from app.config import settings

router = APIRouter(tags=["health"])


@router.get("/healthz")
async def healthz():
    return {"status": "ok", "service": "auditgraph", "version": settings.PIPELINE_VERSION}


@router.get("/readyz")
async def readyz():
    return {
        "status": "ready",
        "service": "auditgraph",
        "pipeline_version": settings.PIPELINE_VERSION,
        "database": "sqlite_async",
        "redis": "fallback_in_memory",
    }


@router.get("/api/v1/version")
async def version():
    return {
        "version": settings.PIPELINE_VERSION,
        "scoring_config_version": settings.SCORING_CONFIG_VERSION,
        "detector_config_version": settings.DETECTOR_CONFIG_VERSION,
    }
